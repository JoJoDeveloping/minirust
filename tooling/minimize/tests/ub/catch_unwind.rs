#![feature(core_intrinsics)]
#![allow(internal_features)]

#[allow(unconditional_panic)]
fn try_fn(_data_ptr: *mut u8) {
    assert!(false);
}

#[allow(unconditional_panic)]
fn catch_fn(_data_ptr: *mut u8, _payload: *mut u8) {
    assert!(false);
}

fn main() {
    let mut data: u8 = 0;
    let data_ptr = &mut data as *mut u8;
    unsafe {
        // Both `try_fn` and `catch_fn` panic. This results in UB.
        core::intrinsics::catch_unwind(try_fn, data_ptr, catch_fn);
    }
}
