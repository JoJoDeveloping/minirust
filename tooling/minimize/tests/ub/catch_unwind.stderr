fatal error: UB: unwinding from a function where the caller did not specify an unwind_block
